### PR TITLE
Update DirectoryProperty to use @InputDirectory

### LINKED ISSUE #370: Sentry Android Gradle plugin is not compatible with Gradle 8.0
### Gradle Version

8.0

### AGP Version

any

### Code Minifier/Optimizer

_No response_

### Version

3.1.4

### Sentry SDK Version

any

### Steps to Reproduce

1. `git clone git@github.com:asodja/io-sentry-with-gradle8.git`
2. `./gradlew :validatePluginWithId_io_sentry_android_gradle`

### Expected Result

The build is successful.

### Actual Result

Build fails with:
```
Execution failed for task ':validatePluginWithId_io_sentry_android_gradle'.
> Plugin validation failed with 2 problems:
    - Error: Type 'io.sentry.android.gradle.tasks.SentryUploadNativeSymbolsTask' property 'buildDir' has @Input annotation used on property of type 'DirectoryProperty'.
      
      Reason: A property of type 'DirectoryProperty' annotated with @Input cannot determine how to interpret the file.
      
      Possible solution: Annotate with @InputDirectory for directories.
      
      Please refer to https://docs.gradle.org/7.7-20220819221134+0000/userguide/validation_problems.html#incorrect_use_of_input_annotation for more details about this problem.
    - Error: Type 'io.sentry.android.gradle.tasks.SentryUploadProguardMappingsTask' property 'uuidDirectory' has @Input annotation used on property of type 'DirectoryProperty'.
      
      Reason: A property of type 'DirectoryProperty' annotated with @Input cannot determine how to interpret the file.
      
      Possible solution: Annotate with @InputDirectory for directories.
      
      Please refer to https://docs.gradle.org/7.7-20220819221134+0000/userguide/validation_problems.html#incorrect_use_of_input_annotation for more details about this problem.
  See https://docs.gradle.org/7.7-20220819221134+0000/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
```

Context: Gradle 8 will have more strict validation for annotations applied to file/directory types: https://github.com/gradle/gradle/pull/21153.

### PR DESCRIPTION
**Pull request by cortinico**
## :scroll: Description
Fixes #370

## :bulb: Motivation and Context
See #370 for further context. This is needed for Gradle 8 support.

## :green_heart: How did you test it?
Will rely on CI tests.

## :pencil: Checklist
- [x] I reviewed the submitted code
- [x] I updated the docs if needed
- [x] No breaking changes

#skip-changelog

### CODE REVIEW

**Comment by adinauer** [comment]
Thanks for the PR @cortinico. I was gonna add a changelog but saw that you put #skip-changelog. Any reason this shouldn't be mentioned in the changelog?

**Comment by cortinico** [comment]
> Any reason this shouldn't be mentioned in the changelog?

I'm fine either way. The reason I haven't added a changelog is because this is not a user facing change so it's probably not relevant for users. Up to you

**Comment by adinauer** [comment]
@cortinico the way I understand this is that people using our plugin and also using Gradle 8 have their build failed by the `@Input` annotation. Sounds to me like it's user facing. Is my understanding wrong?

I'll add the changelog.

**Comment by cortinico** [comment]
> the way I understand this is that people using our plugin and also using Gradle 8 have their build failed by the `@Input` annotation

Sorry I wasn't exact.
This is one of the things to fix for Gradle 8 support. However there are other necessary changes to do (e.g. bump the Kotlin version). So mentioning in the release notes that this fixes the plugin for Gradle 8 is misleading.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@
   - [changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md#641)
   - [diff](https://github.com/getsentry/sentry-java/compare/6.4.0...6.4.1)
 
+### Fixes
+
+- Update DirectoryProperty to use @InputDirectory so the build no longer fails when using Gradle 8 ([#374](https://github.com/getsentry/sentry-android-gradle-plugin/pull/374))
```
Comment (adinauer) [inline CHANGELOG.md:13]:
```suggestion
- Update DirectoryProperty to use @InputDirectory ([#374](https://github.com/getsentry/sentry-android-gradle-plugin/pull/374))
```

**Review by adinauer — APPROVED** [review]

**Pull request merged**